keanu-project/src/main/java/io/improbable/keanu/vertices/intgrtensor/IntegerVertex.java

```diff
@@ -0,0 +1,133 @@
+package io.improbable.keanu.vertices.intgrtensor;
+
+
+import io.improbable.keanu.kotlin.IntegerOperators;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.DiscreteTensorVertex;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.CastIntegerVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.ConstantIntegerVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.operators.binary.IntegerAdditionVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.operators.binary.IntegerDifferenceVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.operators.binary.IntegerDivisionVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.operators.binary.IntegerMultiplicationVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.operators.unary.IntegerAbsVertex;
+import io.improbable.keanu.vertices.intgrtensor.nonprobabilistic.operators.unary.IntegerUnaryOpLambda;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public abstract class IntegerVertex extends DiscreteTensorVertex<IntegerTensor> implements IntegerOperators<io.improbable.keanu.vertices.intgrtensor.IntegerVertex> {
+
+    public io.improbable.keanu.vertices.intgrtensor.IntegerVertex minus(io.improbable.keanu.vertices.intgrtensor.IntegerVertex that) {
```

Comment:
Why do we have the full class path here? 

`io.improbable.keanu.vertices.intgrtensor.IntegerVertex`

Reply:
Looks like some confusion between tensor IntegerVertex and non-tensor IntegerVertex. Clean it up. Good catch.